### PR TITLE
Change cronet interface

### PR DESCRIPTION
**Pull request by gcasto**
This change renames and moves cronet_c_for_grpc.h to bidirectional_stream_c.h and updates class names to mirror a change in Chromium that is currently under review (https://codereview.chromium.org/2470923002/). This change should land shortly after the Chromium change to prevent breakage of Cronet in gRPC.

### CODE REVIEW

**Comment by gcasto** [comment]
Given the way the Cronet tests work, I need to land the Chromium CL first before these changes become relevant. Unfortunately that means that there is going to be some period of time where the Cronet tests in gRPC are going to break (between the Chromium CL landing, and this CL landing). Not sure a way around that.

**Comment by gcasto** [comment]
I've verified that this works locally with a Cronet.framework built from Chromium after https://codereview.chromium.org/2492703002/.

**Comment by gcasto** [comment]
Ping?

**Comment by gcasto** [comment]
@muxi, I don't seem to have the ability to assign reviewers.

**Comment by muxi** [comment]
test this please

**Comment by gcasto** [comment]
Looks like there has been some additional code added which references the old names in cronet_c_for_grpc.h. Updating those now. The rest of the failures look unrelated to me.

**Comment by muxi** [comment]
Thanks Garrette.

**Comment by muxi** [comment]
test this please

**Comment by gcasto** [comment]
Most of these don't look like real test failures to me. I'm going to try re-running these locally, but none of these look cronet related.

**Comment by gcasto** [comment]
Merged again and updated build file. I'm actually not sure how to see the test run from the previous push, but testing locally only got one repeatable failure (bins/opt/lb_policies_test), which I think is unrelated. Other tests failed occasionally depending on the run. I'll test against head to verify, but it seems like the tests are just flaky.

**Comment by muxi** [comment]
test this please

File: BUILD [hunk 1]
```diff
@@ -1521,6 +1521,151 @@ cc_library(
     "src/cpp/server/dynamic_thread_pool.h",
     "src/cpp/server/thread_pool_interface.h",
     "src/cpp/thread_manager/thread_manager.h",
+    "src/core/ext/transport/chttp2/transport/bin_decoder.h",
```
Comment (muxi) [inline BUILD]: These seems to be unrelated changes?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1259,6 +1259,180 @@ add_library(grpc++_cronet
   src/cpp/util/string_ref.cc
   src/cpp/util/time_cc.cc
   src/cpp/codegen/codegen_init.cc
+  src/core/ext/transport/chttp2/client/insecure/channel_create.c
```
Comment (muxi) [inline CMakeLists.txt]: Seems unrelated too

File: Makefile [hunk 3]
```diff
@@ -3978,6 +3978,180 @@ LIBGRPC++_CRONET_SRC = \
     src/cpp/util/string_ref.cc \
     src/cpp/util/time_cc.cc \
     src/cpp/codegen/codegen_init.cc \
+    src/core/ext/transport/chttp2/client/insecure/channel_create.c \
```
Comment (muxi) [inline Makefile]: And also seems unrelated...

**Comment by muxi** [comment]
CoreCronetEnd2EndTests.m requires changes as well

**Comment by gcasto** [comment]
Updated CoreCronetEnd2EndTests last week.

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
It seems cronet_transport.c needs a run of `clang-format`. Other than that the changes LGTM.

C/C++ are running a test greening session and frozen C/C++ portions of master. I need to check with them on when we can merge this in.

**Comment by muxi** [comment]
cc: @makdharma

**Comment by gcasto** [comment]
Ran clang-format

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
Still waiting for Fixit to finish..

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
@nicolasnoble - The `generate_projects.sh` test fails in `Basic Tests Linux` despite that I have run it locally without creating any new diff. Checked all submodules on my local are ok. Do you know what I need to do to make it pass?
========== Failure message ===========
897,905c897
< doc/wait-for-ready.md \
< doc/stress_test_framework.md \
< doc/binary-logging.md \
< doc/load-balancing.md \
< doc/g_stands_for.md \
< doc/connectivity-semantics-and-api.md \
< doc/negative-http2-interop-test-descriptions.md \
< doc/c-style-guide.md \
< doc/fail_fast.md \
---
> doc/statuscodes.md \
907,908c899
< doc/server_reflection_tutorial.md \
< doc/server-reflection.md \
---
> doc/g_stands_for.md \
910,914c901
< doc/compression_cookbook.md \
< doc/health-checking.md \
< doc/http-grpc-status-mapping.md \
< doc/connection-backoff-interop-test-description.md \
< doc/command_line_tool.md \
---
> doc/server_reflection_tutorial.md \
915a903,904
> doc/health-checking.md \
> doc/c-style-guide.md \
916a906,911
> doc/connectivity-semantics-and-api.md \
> doc/load-balancing.md \
> doc/command_line_tool.md \
> doc/wait-for-ready.md \
> doc/connection-backoff-interop-test-description.md \
> doc/server-reflection.md \
919,921c914
< doc/PROTOCOL-HTTP2.md \
< doc/statuscodes.md \
< doc/epoll-polling-engine.md \
---
> doc/negative-http2-interop-test-descriptions.md \
922a916,922
> doc/fail_fast.md \
> doc/compression_cookbook.md \
> doc/binary-logging.md \
> doc/epoll-polling-engine.md \
> doc/stress_test_framework.md \
> doc/http-grpc-status-mapping.md \
> doc/PROTOCOL-HTTP2.md \
Traceback (most recent call last):
  File "tools/buildgen/generate_projects.py", line 111, in <module>
    assert 0 == os.system('diff %s %s' % (s, g)), s
AssertionError: ./tools/doxygen/Doxyfile.c++.internal

FAILED: tools/buildgen/generate_projects.sh [ret=1, pid=1483]

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
This is ready for merge. Do it tomorrow.

**Comment by muxi** [comment]
test this please

**Comment by muxi** [comment]
Test failures:
Bazel Full Build & Tests - #9379
Basic Tests Linux - #9380, #9381
